core/node/node_framework/src/implementations/layers/base_token_adjuster.rs

```diff
@@ -0,0 +1,71 @@
+use zksync_config::configs::base_token_adjuster::BaseTokenAdjusterConfig;
+use zksync_dal::Core;
+use zksync_db_connection::connection_pool::ConnectionPool;
+use zksync_types::Address;
+
+use crate::{
+    implementations::resources::pools::{MasterPool, PoolResource},
+    service::{ServiceContext, StopReceiver},
+    task::{Task, TaskId},
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+/// A layer that wires the Base Token Adjuster task.
+#[derive(Debug)]
+pub struct BaseTokenAdjusterLayer {
+    base_token_l1_address: Address,
+    config: BaseTokenAdjusterConfig,
+}
+
+impl BaseTokenAdjusterLayer {
+    pub fn new(base_token_l1_address: Address, config: BaseTokenAdjusterConfig) -> Self {
+        Self {
+            base_token_l1_address,
+            config,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for BaseTokenAdjusterLayer {
+    fn layer_name(&self) -> &'static str {
+        "base_token_adjuster"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        let master_pool_resource = context.get_resource::<PoolResource<MasterPool>>().await?;
+        let master_pool = master_pool_resource.get().await?;
+
+        context.add_task(Box::new(BaseTokenAdjusterTask {
+            main_pool: master_pool,
+            base_token_l1_address: self.base_token_l1_address,
+            config: self.config,
+        }));
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct BaseTokenAdjusterTask {
+    main_pool: ConnectionPool<Core>,
+    base_token_l1_address: Address,
+    config: BaseTokenAdjusterConfig,
+}
+
+#[async_trait::async_trait]
+impl Task for BaseTokenAdjusterTask {
+    fn id(&self) -> TaskId {
+        "base_token_adjuster".into()
+    }
+
+    async fn run(self: Box<Self>, stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        let mut adjuster = zksync_base_token_adjuster::MainNodeBaseTokenAdjuster::new(
+            self.main_pool,
+            self.config,
+            self.base_token_l1_address,
+        );
+
+        adjuster.run(stop_receiver.0).await
```

Comment:
No need to run if `base_token_l1_address` is `ETH` 

Could print a warning if its the case

Reply:
This would introduce many more conditionals to the flow. Running this component with ETH as base token would hold a 1:1 conversion ratio.

Reply:
`This would introduce many more conditionals to the flow.`
Could you explain that part?

This component will run in the background and will insert the same entry every 30s (default) in DB when none of that is required for ETH-based chain. Not to mention the house keeper that also need to purge the records. 

At least please add some documentation to make sure operators does not run it when it is not necessary

Reply:
Resolving this as with the new flow the persisting layer isn't added if the operated chain has ETH as a base token and the added resource will be a NOOP resource in this case.